sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3307,12 +3307,18 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       _.containsPattern(NATURAL_LIKE_JOIN), ruleId) {
       case j @ Join(left, right, UsingJoin(joinType, usingCols), _, hint)
           if left.resolved && right.resolved && j.duplicateResolved =>
-        commonNaturalJoinProcessing(left, right, joinType, usingCols, None, hint)
+        val p = commonNaturalJoinProcessing(left, right, joinType, usingCols, None, hint)
+        j.getTagValue(LogicalPlan.PLAN_ID_TAG)
+          .foreach(p.setTagValue(LogicalPlan.PLAN_ID_TAG, _))
```

Comment:
should we reset the plan id to the new join or the new project?

Reply:
I'm not 100% sure about this.

it seems that setting the plan id to new Join also works

Reply:
shall we update the code inside `commonNaturalJoinProcessing` and retain the id in `Join`? That seems more natural.

Reply:
got it